armv7m7-imxrt106x/syspage.c

```diff
@@ -436,35 +439,48 @@ void syspage_addEntries(u32 start, u32 sz)
 int syspage_addProg(void *start, void *end, const char *imap, const char *dmap, const char *name)
 {
 	u8 imapID, dmapID;
-	unsigned int pos = 0, len = 0;
+	unsigned int pos, len;
 	u32 progID = syspage_common.progsCnt;
 
 	if ((syspage_getMapID(imap, &imapID) < 0) || (syspage_getMapID(dmap, &dmapID) < 0)) {
 		plostd_printf(ATTR_ERROR, "\nMAPS for %s doesn not exist!\n", name);
 		return -1;
 	}
 
+	len = plostd_strlen(name);
+
+	if (syspage_common.argCnt + 1 + len + 1 + 1 > MAX_ARGS_SIZE) {
```

Comment:
Sorry, but I think I may have made a mistake.
The `argsCnt` points to `\0'`so you need an extra 2 bytes of `'X'` + `' '` not 3.
Check it again.

Reply:
If you mean `argCnt +  len + 2 >= MAX_ARGS_SIZE`  than `argCnt + len + 3 > MAX_ARGS_SIZE` is the equivalent, but I prefer the current version as it is more readable `X` + `len` + ` ` + `\0`.
Please see the attached figure, for an explanation:

![qw2e59](https://user-images.githubusercontent.com/141153/110604442-6251de80-8188-11eb-967f-cb63a55cc681.png)